pages/curriculum/[lesson].tsx

```diff
@@ -1,22 +1,34 @@
 import * as React from 'react'
 import { useRouter } from 'next/router'
-import useSession from '../../helpers/useSession'
 import withQueryLoader, {
   WithQueryProps
 } from '../../containers/withQueryLoader'
 import Layout from '../../components/Layout'
 import LessonTitleCard from '../../components/LessonTitleCard'
 import Alert from '../../components/Alert'
 import ChallengeMaterial from '../../components/ChallengeMaterial'
-import { GET_LESSON } from '../../graphql/queries'
+import { GET_APP } from '../../graphql/queries'
+import { Lesson, LessonStatus } from '../../@types/lesson'
+import { Session } from '../../@types/session'
+import { UserSubmission } from '../../@types/challenge'
 import _ from 'lodash'
 
 const Challenges: React.FC<WithQueryProps> = ({ queryData }) => {
   const {
-    lessonInfo: currentLesson,
-    userSubmissions,
-    lessonStatus: currentLessonStatus
-  } = queryData
+    lessons,
+    session
+  }: { lessons: Lesson[]; session: Session } = queryData
```

Comment:
is typing here necessary if we're already typing the component props withqueryprops? I can see the incentive of it being more explicit for new people to the codebase, but seems kind of unnecessary

Reply:
We need typing if it's not implicitly provided to `TypeScript`. 

The idea of types is so someone doesn't accidentally assigned something of a different type to variable eliminating the possibility of some errors.